src/react/databrowser/objects/__tests__/ObjectList.test.jsx

```diff
@@ -0,0 +1,91 @@
+    });
+
+    it('should render ObjectList with objects', () => {
+        const { component } = reduxMount(<ObjectList objects={List([S3_OBJECT])} bucketName={BUCKET_NAME}
+            prefixWithSlash=''
+            toggled={List()}/>,
+        );
+
+        expect(component.find('Row')).toHaveLength(1);
+    });
+
+    it('should call openObjectUploadModal by clicking on createFolder button', () => {
+        const openObjectUploadModalSpy = jest.spyOn(s3object, 'openObjectUploadModal');
+
+        const { component } = reduxMount(<ObjectList objects={List([S3_OBJECT])} bucketName={BUCKET_NAME}
+            prefixWithSlash=''
+            toggled={List()}/>,
+        );
+
+        component.find('Button#object-list-upload-button').simulate('click');
+        expect(openObjectUploadModalSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call openFolderCreateModal by clicking on createFolder button', () => {
+        const openFolderCreateModalSpy = jest.spyOn(s3object, 'openFolderCreateModal');
+
+        const { component } = reduxMount(<ObjectList objects={List([S3_OBJECT])} bucketName={BUCKET_NAME}
+            prefixWithSlash=''
+            toggled={List()}/>,
+        );
+
+        component.find('Button#object-list-create-folder-button').simulate('click');
+        expect(openFolderCreateModalSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('Delete button should be disable', () => {
+        const { component } = reduxMount(<ObjectList objects={List([S3_OBJECT])} bucketName={BUCKET_NAME}
+            prefixWithSlash=''
+            toggled={List()}/>,
+        );
+
+        expect(component.find('Button#object-list-delete-button').prop('disabled')).toBe(true);
+    });
+
+    it('Delete button should be enable and should call openObjectDeleteModal when is pressed', () => {
+        const openObjectDeleteModalSpy = jest.spyOn(s3object, 'openObjectDeleteModal');
+
+        const { component } = reduxMount(<ObjectList objects={List([S3_OBJECT])} bucketName={BUCKET_NAME}
+            prefixWithSlash=''
+            toggled={List([S3_OBJECT])}/>,
+        );
+
+
+        const deleteButton = component.find('Button#object-list-delete-button');
+        expect(deleteButton.prop('disabled')).toBe(false);
+        deleteButton.simulate('click');
+        expect(openObjectDeleteModalSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should select all objects when ticking checkbox square', () => {
+        const toggleAllObjectsSpy = jest.spyOn(s3object, 'toggleAllObjects');
+
+        const { component } = reduxMount(<ObjectList objects={List([S3_OBJECT, SECOND_S3_OBJECT])}
+            bucketName={BUCKET_NAME} prefixWithSlash=''
+            toggled={List()}/>,
+        );
+
+
+        const toggleAllObjectsButton = component.find('th#object-list-table-head-checkbox > input');
+        toggleAllObjectsButton.simulate('change');
+        expect(toggleAllObjectsSpy).toHaveBeenCalledTimes(1);
```

Comment:
Is it possible to check that  `S3_OBJECT` and `SECOND_S3_OBJEC`'s checkboxes are toggled/checked?

Reply:
From the component ObjectList it not possible to check that because it doesn't have the control of each objects state

Reply:
A simple test will be to provide `ObjectList` with a list of two objects: 
one with the `toggled` property sets to true and the other sets to false. 
We then can check that the checkbox of the first object is checked and that the second one is not.
Sonething like that `expect(component.find('tbody input[type="checkbox"]').first().prop('checked')).toBe(false/true)`